### ISSUE TITLE
networkgraph: directed edges, digraph, multigraph, self-loop

### ISSUE THREAD

**Issue by vitaliy-kretsul** [highcharts/highcharts#18059]
Hey guys!
Any plans on extending your force-directed networkgraph chart to support any of these features:
- directed edges (edges with arrows)
- digraph/multigraph (two vertices can be connected by multiple edges)
- self-loop (loop-like edges connecting a vertex to itself)

Thanks!

**Comment by karolkolodziej** [comment]
Hi @vitaliy-kretsul!

_1. directed edges (edges with arrows)_
You can modify the `getLinkPath` method to get a different link path- [demo](https://jsfiddle.net/BlackLabel/tfhja8se/).

_2. digraph/multigraph (two vertices can be connected by multiple edges)_
Could you please elaborate or show in an example how it should look like?

_3. self-loop (loop-like edges connecting a vertex to itself)_
I presume it is just a matter of data. Please check the demo from point 1.

**Comment by vitaliy-kretsul** [comment]
Hi @karolkolodziej !

Thanks for an advice! I've seen this demo on forum. Of course that is possible to do, but it's not very convenient and universal. That is more a work-around. What I am requesting is this to be a built-in feature. The three features I listed are very useful and necessary for a graph. I would even dare to assume that they are critical to have in a graph :)
Please, refer to this image to see how there features look like:
![example](https://user-images.githubusercontent.com/12032710/205003359-f84dbc6c-2a0a-4c4e-be09-ae6e0f41bf38.png)

**Comment by karolkolodziej** [comment]
@vitaliy-kretsul Thank you for clarifying!

**Comment by karolkolodziej** [comment]
Nothing has been decided yet, there is no ETA for this ticket.

**Comment by vitaliy-kretsul** [comment]
@karolkolodziej 
yes, for sure. I am not requesting any ETAs, since I understand it's a big feature to implement. Just wandering if the verdict is a 'yes' :)

**Comment by kogratte** [comment]
Hey @karolkolodziej :) I can beg for this feature right now too :)

**Comment by karolkolodziej** [comment]
Hi! 
Thank you for bringing our attention once again to this one! 

Below please find two proofs of concept: 
1. Self-loop- [demo](https://jsfiddle.net/BlackLabel/tfhja8se/2/).
2. Multiple connections- [demo](https://jsfiddle.net/BlackLabel/tfhja8se/4/).

**Comment by kogratte** [comment]
Looks promising. Will try. Thanks for the quick feedback.
